platform-api/src/main/java/io/datacater/core/deployment/K8Deployment.java

```diff
@@ -256,6 +250,19 @@ public Map<String, Object> getDeployment(UUID deploymentId) {
     }
   }
 
+  public Deployment getDeploymentObject(UUID deploymentId) throws KubernetesClientException {
+    try {
+      return client
+          .apps()
+          .deployments()
+          .inNamespace(StaticConfig.EnvironmentVariables.NAMESPACE)
+          .withName(getDeploymentName(deploymentId))
+          .get();
+    } catch (KubernetesClientException e) {
+      throw new DeploymentNotFoundException(StaticConfig.LoggerMessages.DEPLOYMENT_NOT_FOUND);
```

Comment:
There might be other reasons for a `KubernetesClientException` besides the non-existence of the requested deployment. With the current implementation, we lose this information.

Could we either log the message of the `KubernetesClientException` before re-throwing it as a `DeploymentNotFoundException` or pass the message to the `DeploymentNotFoundException`?